aws/resource_aws_lambda_function.go

```diff
@@ -200,9 +200,21 @@ func resourceAwsLambdaFunction() *schema.Resource {
 
 			"tags": tagsSchema(),
 		},
+
+		CustomizeDiff: updateComputedAttributesOnPublish,
 	}
 }
 
+func updateComputedAttributesOnPublish(d *schema.ResourceDiff, meta interface{}) error {
+	publish := d.Get("publish").(bool)
+	if publish && needsFunctionCodeUpdate(d) {
```

Comment:
What if you did the following.  It would make sure the last_modified was updated even if publish is set to false.

```
	if needsFunctionCodeUpdate(d) {
		d.SetNewComputed("last_modified")

		if publish {
			d.SetNewComputed("version")
			d.SetNewComputed("qualified_arn")
		}
	}
```

Reply:
Check out the new commit. It should cover your scenario now too